contracts/L2/ETHxPoolV5.sol

```diff
@@ -0,0 +1,171 @@
+    /// @param _bridger The bridger address
+    /// @param _ethx The ETHx token address
+    /// @param _feeBps The fee basis points
+    /// @param _ethxOracle The ethxOracle address
+    function initialize(
+        address _admin,
+        address _bridger,
+        address _ethx,
+        uint256 _feeBps,
+        address _ethxOracle
+    )
+        public
+        initializer
+    {
+        _checkNonZeroAddress(_ethx);
+        _checkNonZeroAddress(_ethxOracle);
+
+        __AccessControl_init();
+        __ReentrancyGuard_init();
+
+        _grantRole(DEFAULT_ADMIN_ROLE, _admin);
+        _setupRole(BRIDGER_ROLE, _bridger);
+
+        ETHx = IERC20(_ethx);
+        feeBps = _feeBps;
+        ethxOracle = _ethxOracle;
+    }
+
+    /// @dev Swaps ETH for ETHx
+    /// @param _referralId The referral id
+    function deposit(string memory _referralId) external payable nonReentrant {
+        uint256 amount = msg.value;
+
+        if (amount == 0) revert InvalidAmount();
+
+        (uint256 ethxAmount, uint256 fee) = viewSwapETHxAmountAndFee(amount);
+
+        feeEarnedInETH += fee;
+
+        ETHx.safeTransfer(msg.sender, ethxAmount);
+
+        emit SwapOccurred(msg.sender, ethxAmount, fee, _referralId);
+    }
+
+    /// @dev view function to get the ETHx amount for a given amount of ETH
+    /// @param _amount The amount of ETH
+    /// @return ethxAmount The amount of ETHx that will be received
+    /// @return fee The fee that will be charged
+    function viewSwapETHxAmountAndFee(uint256 _amount) public view returns (uint256 ethxAmount, uint256 fee) {
+        fee = _amount * feeBps / 10_000;
+        uint256 amountAfterFee = _amount - fee;
+
+        (, int256 ethxToEthRate,,,) = AggregatorV3Interface(ethxOracle).latestRoundData();
+
+        // Calculate the final ETHx amount
+        ethxAmount = amountAfterFee * ETHX_BASE_RATE / uint256(ethxToEthRate);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            ACCESS RESTRICTED FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @dev Withdraws fees earned by the pool
+    function withdrawFees(address _receiver) external onlyRole(BRIDGER_ROLE) {
+        // withdraw fees in ETH
+        uint256 amountToSendInETH = feeEarnedInETH;
+        feeEarnedInETH = 0;
+        (bool success,) = payable(_receiver).call{ value: amountToSendInETH }("");
+        if (!success) revert TransferFailed();
+
+        emit FeesWithdrawn(amountToSendInETH);
+    }
+
+    /// @dev Withdraws assets from the contract for bridging
+    function moveAssetsForBridging() external onlyRole(BRIDGER_ROLE) {
+        // withdraw ETH - fees
+        uint256 ethBalanceMinusFees = address(this).balance - feeEarnedInETH;
+
+        (bool success,) = msg.sender.call{ value: ethBalanceMinusFees }("");
+        if (!success) revert TransferFailed();
+
+        emit AssetsMovedForBridging(ethBalanceMinusFees);
+    }
+
```

Comment:
Need a fn to withdraw provisioned ETHx if we ever want to downsize the pool. Should be manager / bridger role. 

Given this fn, i would rename bridger to manager. Because it does a few admin fns as well and not just bridging.

Reply:
[ac52ee9](https://github.com/stader-labs/ethx_oft/pull/8/commits/ac52ee93635d1c77b83649a9598ab7c355519243)